test/spec/common/client/set-cache.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+var test = require('tape');
+
+test('cache is not empty object when specified in opts', function(t) {
+  t.plan(1);
+
+  var opts = {
+    cache: {
+      'path/for/query': {}
+    }
+  };
+
+  var algoliasearch = require('../../../../');
+  var search = algoliasearch('applicationID', 'apiKey', opts);
+
+  t.deepEqual(
+    search.cache,
```

Comment:
The best way to test this is to actually initialize a client with some fake app id and index then check that when doing a query you will get the cached response you provided, can you do so?